pkg/didcomm/protocol/introduce/service.go

```diff
@@ -68,16 +69,26 @@ type Service struct {
 	service.Message
 	store     storage.Store
 	callbacks chan callback
+	ctx       internalContext
+}
+
+// provider contains dependencies for the DID exchange protocol and is typically created by using aries.Context()
+type provider interface {
+	OutboundDispatcher() dispatcher.Outbound
+	StorageProvider() storage.Provider
 }
 
 // New returns introduce service
-func New(ctx context.Context, s storage.Provider) (*Service, error) {
-	store, err := s.OpenStore(Introduce)
+func New(ctx context.Context, p provider) (*Service, error) {
```

Comment:
Std lib context.Context should be provided to a request not a constructor.

Reply:
I have created it to stop the listener 
https://github.com/hyperledger/aries-framework-go/pull/575/files#diff-43cf3af5d3c6b9acebc5c635ba4d0f7dR97

Reply:
usage here https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/introduce/service.go#L110

Reply:
This seems to be an usual usage of context. Recommend a Stop method instead.

Reply:
Done!